### DISCUSSION TITLE
DAO not declared within the daos list of the database

### DISCUSSION THREAD

**Discussion by KarrarMohamed** [simolus3/drift#1987] [Q&A] [answered]
I'm currently working on a project and using moor as may main database

I'm using Riverpod as the state management for my project

```dart
@DriftDatabase(tables: [
  Users,
  Employess, .. etc
],
daos: [ ]
)
`class MyAppDatabase extends _$MyAppDatabase {
  MyAppDatabase() : super(_openConnection());
  @override
  int get schemaVersion => 2;
}
```

I'm using daos and the database  as a singletons objects by the aid of riverpod 

```dart
/// database singleton
final driftDbProvider = Provider<MyAppDatabase>((ref) => MyAppDatabase());`

// dao singleton
final someDaoProvider = Provider<SomeDao>((ref) {
  return SomeDao(db);
}); 
```

but I need to pass a parameter (ref) to dao so I can do more with it
as below

```dart
final someDaoProvider = Provider<SomeDao>((ref) {
  final db = ref.watch(driftDbProvider);
  return SomeDao(db, ref);
});
```

the question is here
when I do not pass any parameter to the dao object, everything works well and the code generation works fine

But 

when I pass the ref to the dao object, the code generation end successfully but still there are a compile time errors in the generated file of the main db

this occur when I declare the dao within the database class

```dart
@DriftDatabase(tables: [
  Users,
  Employess, .. etc
],
daos: [
SomeDao , 
........
 ]
)
`class MyAppDatabase extends _$MyAppDatabase {
  MyAppDatabase() : super(_openConnection());
  @override
  int get schemaVersion => 2;
}
```

But when I do not daclare the dao (that I want it to receive a parameter ) within the list of daos in the database , everything works well and the code generation also works fine and everything is fucntional and work as expected

the question is that

what would happen if I do not declare the daos within the databse daos list

is there any bugs that might appear in the future ??

**Comment by simolus3** [comment, marked as answer]
> is there any bugs that might appear in the future ??

Not really, your approach with riverpod providers sounds fine. Adding them to the main database is just so that it generates accessors for them (so you can do `db.someDao`). If you don't want to make use of that (and in your case, you don't really need to), there's no harm in omitting them from the database's annotation.

**Reply by KarrarMohamed** [reply]
Thanks alot for you for this response and for your amazing package
God bless you
